### ISSUE TITLE
Allow compilation on OSX/macOS

### ISSUE THREAD

**Issue by djui** [scraperwiki/pdf2msgpack#21]
Current was output:

```
$ ./waf configure
Setting top to                           : ./pdf2msgpack
Setting out to                           : ./pdf2msgpack/build
Checking for 'clang++' (C++ compiler)    : /usr/bin/clang++
Checking for compiler flags --std=c++14  : yes
Building dynamic binaries                : yes
Checking for program 'pkg-config'        : /usr/local/bin/pkg-config
Checking for 'poppler'                   : yes
Checking linux-headers & seccomp support : not found
The configuration failed
(complete log in ./pdf2msgpack/build/config.log)
```

As discussed, making the seccomp support OS/platform dependant would be a good step forward.

**Comment by pwaller** [comment]
See https://github.com/scraperwiki/pdf2msgpack/pull/22. That should get us past the first failure. Not sure about subsequent ones.

**Comment by pwaller** [comment]
For my reference, what version of clang are you using?

**Comment by djui** [comment]
After #22 and the following small patch, configuration works with:

```
ctx.env.append_value("CXXFLAGS", [
          "-g",
          "-Wall",
+         "-Wno-unused-private-field",
          "-Werror",
          "-ansi",
          "--std=c++14",
      ])
```

```
$ CXXFLAGS="-I/usr/local/Cellar/poppler/0.45.0/include" ./waf configure --disable-syscall-filter
Setting top to                           : /Users/uwe/dev/scraperwiki/pdf2msgpack
Setting out to                           : /Users/uwe/dev/scraperwiki/pdf2msgpack/build
Checking for 'clang++' (C++ compiler)    : /usr/bin/clang++
Checking for compiler flags --std=c++14  : yes
Checking linux-headers & seccomp support : not found
Building dynamic binaries                : yes
Checking for program 'pkg-config'        : /usr/local/bin/pkg-config
Checking for 'poppler'                   : yes
Checking libpoppler-private-dev (poppler configured with --enable-xpdf-headers) : yes
Disable syscall filter (***NOT FOR PRODUCTION***)                               : yes
'configure' finished successfully (0.477s)
```

**Comment by pwaller** [comment]
Woot. And `./waf build`?

**Comment by pwaller** [comment]
(note: once waf configure is done, it's no longer necessary to specify the additional environment).

**Comment by djui** [comment]
Prematurely closed #22. `waf build` fails.
